### PR TITLE
Fixes props being ignored on ListItem

### LINKED ISSUE #1343: ListItem: xxxProps for text component are ignored
see description in https://github.com/react-native-training/react-native-elements/issues/1341#issuecomment-409363792

Problem was introduced with following commit
https://github.com/react-native-training/react-native-elements/commit/f8af41c1959625c1d3679a88ad9437059abe39aa

### PR DESCRIPTION
**Pull request by msevestre**
Fixes #1343 
the defaultProps (e.g. titleProps, rightTitleProps ...etc) where completely ignored after refactoring.
This should fix the issue. Also I have updated the snapshot test to add some custom props

### CODE REVIEW

File: src/list/__tests__/ListItem.js [hunk 1]
```diff
@@ -62,6 +62,7 @@ describe('ListItem component', () => {
     const component = shallow(
       <ListItem
         title="title test"
+        titleProps={{ numberOfLines: 5 }}
```
Comment (msevestre) [inline src/list/__tests__/ListItem.js]: Custom props added here

**Comment by msevestre** [comment]
Ok looking good now

**Comment by martinezguillaume** [comment]
Looks good to me ! Sorry for the mistake 😔

**Comment by msevestre** [comment]
@martinezguillaume @iRoachie Awesome. Thanks for merging. I have this fix in production and it works great
Do you guys know when beta 6 is coming out? 
Thanks again for all your hard work

**Pull request merged**